packages/core/src/modules/credentials/formats/jsonld/JsonLdCredentialFormat.ts

```diff
@@ -15,19 +14,53 @@ export interface JsonCredential {
   [key: string]: unknown
 }
 
-// this is the API interface (only)
-export interface JsonLdSignCredentialFormat {
+/**
+ * Format for creating a jsonld proposal, offer or request.
+ */
+export interface JsonLdCredentialDetailFormat {
   credential: JsonCredential
-  options: JsonLdOptionsRFC0593
+  options: {
+    proofPurpose: string
+    proofType: string
+  }
 }
 
-// use this interface internally as the above may diverge in future
-export interface SignCredentialOptionsRFC0593 {
-  credential: JsonCredential
-  options: JsonLdOptionsRFC0593
+// use empty object in the acceptXXX jsonld format interface so we indicate that
+// the jsonld format service needs to be invoked
+type EmptyObject = Record<string, never>
```

Comment:
Not sure how it would work the modularisation, but would it make sense to abstract these types so we can reuse them elsewhere? Might be a bit overengineerd, but I think we have quite a bit of object types in different files defined as well.

Reply:
Yes, maybe. As we're going to abstract more things out of core I'm more leaning to having some duplication. We now have to duplicate them whenver we extract it out of core, as exporting an `EmptyObject` type doesn't make sense.

I'm not too worried about such simple types